python_modules/dagster/dagster/_core/definitions/metadata/source_code.py

```diff
@@ -139,7 +139,14 @@ def _with_code_source_single_definition(
             }
 
     return AssetsDefinition.dagster_internal_init(
-        **{**assets_def.get_attributes_dict(), "metadata_by_key": metadata_by_key}
+        **{
+            **assets_def.get_attributes_dict(),
+            **{
+                "specs": [
+                    spec._replace(metadata=metadata_by_key[spec.key]) for spec in assets_def.specs
+                ]
+            },
+        }
```

Comment:
A bit odd here that we call `get_attributes_dict` twice. What is the intended business logic here?

Reply:
Are you referring to how we also use it in a different function in the same file?

Reply:
Oh whoops totally misread this. I thought you were calling `get_attributes_dict` twice within the same expression but I mixed up red/green 🙈